Gems/ROS2/Code/Source/RobotImporter/URDF/VisualsMaker.cpp

```diff
@@ -225,36 +226,262 @@ namespace ROS2
+    {
+        auto material = visual->Material();
+
+        if (!material)
+        {
+            // Material is optional, it might not appear on all visuals.
             return;
         }
 
-        AZ::Entity* entity = AzToolsFramework::GetEntityById(entityId);
+        AZ_Assert(material->Element(), "Material has data but no Element pointer. Something unexpected has happened with the SDF parsing.");
 
-        // As a Material doesn't have a name and there can only be 1 material per <visual> tag,
-        // the Visual Name is used for the material
-        const std::string materialName{ visual->Name() };
-        const AZStd::string azMaterialName{ materialName.c_str(), materialName.size() };
+        // Conversions from <material> in the file to O3DE are extremely imprecise because the data is going from a Phong model to PBR,
+        // and there are no direct translations from one type of lighting model to the other. All of the conversions will create some
+        // rough approximations of the source lighting data, but should hopefully provide a reasonable starting point for tuning the look.
 
-        // If present in map, take map color definition as priority, otherwise apply local node definition
-        const auto materialColorUrdf =
-            m_materials.contains(azMaterialName) ? m_materials.at(azMaterialName)->Diffuse() : visual->Material()->Diffuse();
-        const AZ::Color materialColor = URDF::TypeConversions::ConvertColor(materialColorUrdf);
+        // Also, URDF/SDF files don't have a concept of overriding specific materials, so every material override generated
+        // below will get applied to *all* materials for a mesh file.
 
-        entity->CreateComponent(AZ::Render::EditorMaterialComponentTypeId);
-        AZ_Trace("AddVisual", "Setting color for material %s\n", azMaterialName.c_str());
-        entity->Activate();
-        AZ::Render::MaterialComponentRequestBus::Event(
-            entityId,
-            &AZ::Render::MaterialComponentRequestBus::Events::SetPropertyValue,
-            AZ::Render::DefaultMaterialAssignmentId,
-            "settings.color",
-            AZStd::any(materialColor));
-        entity->Deactivate();
+        // First, force the model asset to get loaded into memory before adding the material component.
+        // This is required so that we can get the default material map that will be used to override properties for each material.
+        auto modelAsset = AZ::Data::AssetManager::Instance().GetAsset<AZ::RPI::ModelAsset>(assetId, AZ::Data::AssetLoadBehavior::Default);
+        modelAsset.BlockUntilLoadComplete();
+
+        // Initialize the material component configuration to contain all of the material mappings from the model.
+        AZ::Render::MaterialComponentConfig config;
+        config.m_materials = AZ::Render::GetDefautMaterialMapFromModelAsset(modelAsset);
```

Comment:
We should check if the Model Asset is ready before proceding.

Reply:
I'm kinda mixed on that. Even if the model fails to load, we _could_ still continue and set the default materials without the slots to the override values, and create the material component. It would be reasonable to print an error if it failed to load though.

Reply:
Added an error message.